qiita_db/support_files/patches/python_patches/51.py

```diff
@@ -0,0 +1,107 @@
+from future.utils import viewitems
+from datetime import datetime
+
+from qiita_db.metadata_template.constants import (
+    SAMPLE_TEMPLATE_COLUMNS, PREP_TEMPLATE_COLUMNS,
+    PREP_TEMPLATE_COLUMNS_TARGET_GENE)
+from qiita_db.metadata_template.prep_template import PrepTemplate
+from qiita_db.metadata_template.sample_template import SampleTemplate
+from qiita_db.sql_connection import TRN
+
+
+# getting columns in each info file that we need to check for
+cols_sample = [col
+               for key, vals in viewitems(SAMPLE_TEMPLATE_COLUMNS)
+               for col, dt in viewitems(vals.columns) if dt == datetime]
+cols_prep = [col
+             for key, vals in viewitems(PREP_TEMPLATE_COLUMNS)
+             for col, dt in viewitems(vals.columns) if dt == datetime].extend(
+                [col
+                 for key, vals in viewitems(PREP_TEMPLATE_COLUMNS_TARGET_GENE)
+                 for col, dt in viewitems(vals.columns)])
+
+
+def transform_date(value):
+    # for the way the patches are applied we need to have this import and
+    # the next 2 variables within this function
+    from datetime import datetime
+
+    old_formats = [
+        # 4 digits year
+        '%m/%d/%Y %H:%M:%S', '%m-%d-%Y %H:%M',
+        '%m/%d/%Y %H', '%m-%d-%Y', '%m-%Y', '%Y',
+        # 2 digits year
+        '%m/%d/%y %H:%M:%S', '%m-%d-%y %H:%M',
+        '%m/%d/%y %H', '%m-%d-%y', '%m-%y', '%y'
+        ]
+
+    new_formats = [
+        # 4 digits year
+        '%Y-%m-%d %H:%M:%S', '%Y-%m-%d %H:%M',
+        '%Y-%m-%d %H', '%Y-%m-%d', '%Y-%m', '%Y'
+    ]
+
+    # loop over the old formats to see which one is it
+    date = None
+    for i, fmt in enumerate(old_formats):
+        try:
+            date = datetime.strptime(value, fmt)
+            break
+        except ValueError:
+            pass
+    if date is not None:
```

Comment:
Any suggestions? Basically, is testing which is the fmt of the date within the old formats, if it's found it using the new format. Now, new format doesn't accept 2 digits so I'm simply reassigning to one with 4 digits if one of the 2 year digits is found ...